WordPress/Classes/Utility/Logging/WPCrashLoggingProvider.swift

```diff
@@ -23,4 +32,89 @@ class WPCrashLoggingProvider: CrashLoggingDataProvider {
 
         return TracksUser(userID: account.userID.stringValue, email: account.email, username: account.username)
     }
+
+    var loggingUploadDelegate: EventLoggingDelegate {
+        return self
+    }
+}
+
+extension WPCrashLoggingProvider: EventLoggingDelegate {
+    var shouldUploadLogFiles: Bool {
+        return
+            !ProcessInfo.processInfo.isLowPowerModeEnabled
+            && !self.userHasOptedOut
+    }
+
+    var shouldUploadLogFilesForNonFatalEvents: Bool {
+        return true
+    }
+
+    func didQueueLogForUpload(_ log: LogFile) {
+        DDLogDebug("📜 Added log to queue: \(log.uuid)")
+
+        if let eventLogging = CrashLogging.eventLogging {
+            DDLogDebug("📜\t There are \(eventLogging.queuedLogFiles.count) logs in the queue.")
+        }
+    }
+
+    func didStartUploadingLog(_ log: LogFile) {
+        DDLogDebug("📜 Started uploading encrypted log: \(log.uuid)")
+    }
+
+    func didFinishUploadingLog(_ log: LogFile) {
+        DDLogDebug("📜 Finished uploading encrypted log: \(log.uuid)")
+        if let eventLogging = CrashLogging.eventLogging {
+            DDLogDebug("📜\t There are \(eventLogging.queuedLogFiles.count) logs remaining in the queue.")
+        }
+    }
+
+    func uploadFailed(withError error: Error, forLog log: LogFile) {
+        DDLogError("📜 Error uploading encrypted log: \(log.uuid)")
+        DDLogError("📜\t\(error.localizedDescription)")
+
+        let nserror = error as NSError
+        DDLogError("📜\t Code: \(nserror.code)")
+        if let details = nserror.localizedFailureReason {
+            DDLogError("📜\t Details: \(details)")
+        }
+    }
+}
+
+struct EventLoggingDataProvider: EventLoggingDataSource {
+
+    typealias LogFilesCallback = (() -> [URL])
+
+    /// A block that returns all existing log files
+    private let fetchLogFiles: LogFilesCallback?
+
+    /// Initialize the data provider using a block.
+    ///
+    /// Because the most recent log file path can change at runtime (for example; if a given session spans a day boundary the logging system may roll the
+    /// log file transparently in the background) we must determine which is the most recent log file each time we access it.
+    init(_ block: @escaping LogFilesCallback) {
+        self.fetchLogFiles = block
+    }
+
+    /// The key used to encrypt log files
+    let loggingEncryptionKey: String = ApiCredentials.encryptedLogKey()
+
+    /// The previous session log will be the most recent one, because they're split by day
+    var previousSessionLogPath: URL? {
+        return fetchLogFiles?().first
+    }
+
+    /// The current session log will always be the most recent one
+    var currentSessionLogPath: URL? {
+        return fetchLogFiles?().first
+    }
```

Comment:
I have a PR open for this over in https://github.com/Automattic/Automattic-Tracks-iOS/pull/137. 

Once that lands, the body of this will just become `fetchLogFiles?().first` for all cases.